### PR TITLE
ENG-12968 NT-Procedure memory leak fix

### PR DESCRIPTION
**Pull request by xinjiacs**
Use internal NT-Procedure connection handler to invoke NT-Procedures and track the call backs. Otherwise the NT-Procedure callAllNodeNTProcedure API ends up keeping all the reference of NT-Procedure internal invocations and eventually hit OOM.

### CODE REVIEW

File: src/frontend/org/voltdb/LightweightNTClientResponseAdapter.java [hunk 1]
```diff
@@ -59,12 +59,12 @@
 
     private final long m_connectionId;
     private final AtomicLong m_handles = new AtomicLong();
-    private final ConcurrentMap<Long, NTNestedProcedureCallback> m_callbacks = new ConcurrentHashMap<>(2048, .75f, 128);
```
Comment (jhugg) [inline src/frontend/org/voltdb/LightweightNTClientResponseAdapter.java]: Is there ever a reason the more generic, "ProcedureCallback" would be stored here?

Comment (xinjiacs) [inline src/frontend/org/voltdb/LightweightNTClientResponseAdapter.java]: The "NTNestedProcedureCallback" works for NT-Procedure Runner invoking procedure locally. Any NT-Proc invocation of `callAllNodeNTProcedure`  will lose the Procedure Runner information on other nodes. Thus we need generic call back to handle this case.

Comment (akhanzode) [inline src/frontend/org/voltdb/LightweightNTClientResponseAdapter.java]: Whats special about NT ClientResponseAdapter? Whats Lightweight about it?

File: src/frontend/org/voltdb/LightweightNTClientResponseAdapter.java [hunk 2]
```diff
@@ -242,14 +241,18 @@ public long connectionId() {
...
-        if (callback==null) {
+        ProcedureCallback callback = m_callbacks.get(clientHandle);
+        if (callback == null) {
             m_logger.rateLimitedLog(SUPPRESS_INTERVAL, Level.WARN, null,
                     "Could not find caller details for client handle %d. Using internal adapter level connection id", clientHandle);
             return connectionId();
         }
 
-        return callback.getConnectionId(clientHandle);
+        if (callback instanceof NTNestedProcedureCallback == false) {
```
Comment (akhanzode) [inline src/frontend/org/voltdb/LightweightNTClientResponseAdapter.java]: Instanceof is expensive if all are going to be NTNestedProcedureCall why the check?

File: src/frontend/org/voltdb/LightweightNTClientResponseAdapter.java (hunk 1 above)
Comment (xinjiacs) [inline src/frontend/org/voltdb/LightweightNTClientResponseAdapter.java]: You can read the comments:
https://github.com/VoltDB/voltdb/blob/master/src/frontend/org/voltdb/LightweightNTClientResponseAdapter.java#L42

Comment (akhanzode) [inline src/frontend/org/voltdb/LightweightNTClientResponseAdapter.java]: I read it after I commented...this is good.

File: src/frontend/org/voltdb/LightweightNTClientResponseAdapter.java (hunk 2 above)
Comment (xinjiacs) [inline src/frontend/org/voltdb/LightweightNTClientResponseAdapter.java]: The NT-Procedure called from existing NT-Procedure does not have the NTNestedProcedureCallback. You can check at: 
https://github.com/VoltDB/voltdb/blob/abf3b52342122af088bd9b5c48be62df532a9427/src/frontend/org/voltdb/ClientInterface.java#L1208

**Pull request merged**